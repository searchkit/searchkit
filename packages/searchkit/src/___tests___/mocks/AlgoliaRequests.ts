export const SimpleRequest = [
  {
    indexName: 'imdb_movies',
    params: {
      facetFilters: [['type:movie']],
      facets: ['*'],
      highlightPostTag: '</em>',
      highlightPreTag: '<em>',
      maxValuesPerFacet: 10,
      page: 0,
      query: 'shawshank',
      tagFilters: ''
    }
  }
]

export const nonDynamicFacetRequest = [
  {
    indexName: 'imdb_movies',
    params: {
      facets: ['type', 'actors', 'imdbrating'],
      highlightPostTag: '</em>',
      highlightPreTag: '<em>',
      maxValuesPerFacet: 10,
      page: 0,
      query: '',
      tagFilters: ''
    }
  }
]

export const nonDynamicFacetRequestOneFilter = [
  {
    indexName: 'imdb_movies',
    params: {
      facetFilters: [['actors:Alice']],
      facets: ['*'],
      highlightPostTag: '</em>',
      highlightPreTag: '<em>',
      maxValuesPerFacet: 10,
      page: 0,
      query: '',
      tagFilters: ''
    }
  }
]

export const DisjunctiveExampleRequest = [
  {
    indexName: 'imdb_movies',
    params: {
      facetFilters: [['type:movie']],
      facets: ['*'],
      highlightPostTag: '</em>',
      highlightPreTag: '<em>',
      maxValuesPerFacet: 10,
      page: 0,
      query: 'shawshank',
      tagFilters: ''
    }
  },
  {
    indexName: 'imdb_movies',
    params: {
      analytics: false,
      attributesToHighlight: [],
      attributesToRetrieve: [],
      attributesToSnippet: [],
      clickAnalytics: false,
      facets: 'type',
      highlightPostTag: '</em>',
      highlightPreTag: '<em>',
      hitsPerPage: 1,
      maxValuesPerFacet: 10,
      page: 0,
      query: 'shawshank',
      tagFilters: ''
    }
  }
]

export const NestedQueryRequest = [
  {
    indexName: 'imdb_movies',
    params: {
      facetFilters: [['user.first:Alice']],
      facets: ['*'],
      highlightPostTag: '</em>',
      highlightPreTag: '<em>',
      maxValuesPerFacet: 10,
      page: 0,
      query: '',
      tagFilters: ''
    }
  },
  {
    indexName: 'imdb_movies',
    params: {
      analytics: false,
      attributesToHighlight: [],
      attributesToRetrieve: [],
      attributesToSnippet: [],
      clickAnalytics: false,
      facets: 'user.first',
      highlightPostTag: '</em>',
      highlightPreTag: '<em>',
      hitsPerPage: 1,
      maxValuesPerFacet: 10,
      page: 0,
      query: '',
      tagFilters: ''
    }
  }
]

export const TwoFiltersNestedQueryRequest = [
  {
    indexName: 'imdb_movies',
    params: {
      facetFilters: [['user.first:Alice', 'user.first:John']],
      facets: ['*'],
      highlightPostTag: '</em>',
      highlightPreTag: '<em>',
      maxValuesPerFacet: 10,
      page: 0,
      query: '',
      tagFilters: ''
    }
  },
  {
    indexName: 'imdb_movies',
    params: {
      analytics: false,
      attributesToHighlight: [],
      attributesToRetrieve: [],
      attributesToSnippet: [],
      clickAnalytics: false,
      facets: 'user.first',
      highlightPostTag: '</em>',
      highlightPreTag: '<em>',
      hitsPerPage: 1,
      maxValuesPerFacet: 10,
      page: 0,
      query: '',
      tagFilters: ''
    }
  }
]

export const NumericFilterNestedQueryRequest = [
  {
    indexName: 'my-index-000001',
    params: {
      facetFilters: [],
      facets: ['*'],
      highlightPostTag: '</em>',
      highlightPreTag: '<em>',
      maxValuesPerFacet: 10,
      numericFilters: ['user.price>=90'],
      page: 0,
      query: '',
      tagFilters: ''
    }
  }
]

export const FacetQueryExampleRequest = [
  {
    indexName: 'imdb_movies',
    params: {
      facetName: 'type',
      facetQuery: 'm',
      facets: ['*'],
      highlightPostTag: '</em>',
      highlightPreTag: '<em>',
      maxFacetHits: 10,
      maxValuesPerFacet: 10,
      page: 0,
      query: '',
      tagFilters: ''
    }
  }
]

export const NumericFiltersExampleRequest = [
  {
    indexName: 'imdb_movies',
    params: {
      facets: ['type', 'actors', 'imdbrating'],
      highlightPostTag: '</em>',
      highlightPreTag: '<em>',
      maxValuesPerFacet: 10,
      numericFilters: ['imdbrating>=90'],
      page: 0,
      query: '',
      tagFilters: ''
    }
  },
  {
    indexName: 'imdb_movies',
    params: {
      analytics: false,
      clickAnalytics: false,
      facets: 'imdbrating',
      highlightPostTag: '</em>',
      highlightPreTag: '<em>',
      hitsPerPage: 0,
      maxValuesPerFacet: 10,
      page: 0,
      query: ''
    }
  }
]

export const NestedFacetQueryExampleRequest = [
  {
    indexName: 'my-index-000001',
    params: {
      facetName: 'user.first',
      facetQuery: 'har',
      facets: ['user.first', 'user.price'],
      highlightPostTag: '</ais-highlight-0000000000>',
      highlightPreTag: '<ais-highlight-0000000000>',
      maxFacetHits: 10,
      maxValuesPerFacet: 10,
      page: 0,
      query: '',
      tagFilters: ''
    }
  }
]
