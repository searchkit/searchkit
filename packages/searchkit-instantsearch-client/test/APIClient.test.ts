import FrontendClient from '../src/index'
import nock from 'nock'
import { HitsResponseWithFacetFilter } from '../../searchkit/src/___tests___/mocks/ElasticsearchResponses'
import { DisjunctiveExampleRequest } from '../../searchkit/src/___tests___/mocks/AlgoliaRequests'
import 'cross-fetch/polyfill'

describe('API connector', () => {
  it('works as expected', async () => {
    const x = FrontendClient({
      url: 'http://localhost:3000/api/search'
    })

    nock('http://localhost:3000/')
      .post('/api/search', (requestBody: any) => {
        expect(requestBody).toMatchSnapshot('Algolia request')
        return requestBody
      })
      .reply(200, HitsResponseWithFacetFilter)

    const response = await x.search(DisjunctiveExampleRequest as any)

    expect(response).toMatchSnapshot()
    // @ts-ignore
    expect(Object.keys(x.cache)).toMatchInlineSnapshot(`
        [
          "[{"indexName":"imdb_movies","params":{"facetFilters":[["type:movie"]],"facets":["*"],"highlightPostTag":"</em>","highlightPreTag":"<em>","maxValuesPerFacet":10,"page":0,"query":"shawshank","tagFilters":""}},{"indexName":"imdb_movies","params":{"analytics":false,"attributesToHighlight":[],"attributesToRetrieve":[],"attributesToSnippet":[],"clickAnalytics":false,"facets":"type","highlightPostTag":"</em>","highlightPreTag":"<em>","hitsPerPage":1,"maxValuesPerFacet":10,"page":0,"query":"shawshank","tagFilters":""}}]",
        ]
      `)
    x.clearCache()
    // @ts-ignore
    expect(x.cache).toEqual({})
  })
})
