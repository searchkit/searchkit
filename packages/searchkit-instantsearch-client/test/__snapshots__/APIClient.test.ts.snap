// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`API connector works as expected 1`] = `
{
  "responses": [
    {
      "_shards": {
        "failed": 0,
        "skipped": 0,
        "successful": 1,
        "total": 1,
      },
      "aggregations": {
        "actors": {
          "buckets": [
            {
              "doc_count": 1,
              "key": "Bob Gunton",
            },
            {
              "doc_count": 1,
              "key": "Morgan Freeman",
            },
            {
              "doc_count": 1,
              "key": "Tim Robbins",
            },
            {
              "doc_count": 1,
              "key": "William Sadler",
            },
          ],
          "doc_count_error_upper_bound": 0,
          "sum_other_doc_count": 0,
        },
        "rated": {
          "buckets": [
            {
              "doc_count": 1,
              "key": "R",
            },
          ],
          "doc_count_error_upper_bound": 0,
          "sum_other_doc_count": 0,
        },
        "type": {
          "buckets": [
            {
              "doc_count": 1,
              "key": "movie",
            },
          ],
          "doc_count_error_upper_bound": 0,
          "sum_other_doc_count": 0,
        },
      },
      "hits": {
        "hits": [
          {
            "_id": "tt0111161",
            "_index": "imdb_movies",
            "_score": 7.637228,
            "_source": {
              "actors": [
                "Tim Robbins",
                "Morgan Freeman",
                "Bob Gunton",
                "William Sadler",
              ],
              "title": "The Shawshank Redemption",
            },
            "highlight": {
              "desription": [
                "The <em>Shawshank</em> Redemption",
              ],
              "title": [
                "The <em>Shawshank</em> Redemption",
              ],
            },
          },
        ],
        "max_score": 7.637228,
        "total": {
          "relation": "eq",
          "value": 1,
        },
      },
      "status": 200,
      "timed_out": false,
      "took": 2,
    },
    {
      "_shards": {
        "failed": 0,
        "skipped": 0,
        "successful": 1,
        "total": 1,
      },
      "aggregations": {
        "type": {
          "buckets": [
            {
              "doc_count": 1,
              "key": "movie",
            },
          ],
          "doc_count_error_upper_bound": 0,
          "sum_other_doc_count": 0,
        },
      },
      "hits": {
        "hits": [
          {
            "_id": "tt0111161",
            "_index": "imdb_movies",
            "_score": 7.637228,
            "_source": {
              "actors": [
                "Tim Robbins",
                "Morgan Freeman",
                "Bob Gunton",
                "William Sadler",
              ],
              "title": "The Shawshank Redemption",
            },
            "highlight": {
              "description": [
                "The <em>Shawshank</em> Redemption",
              ],
              "title": [
                "The <em>Shawshank</em> Redemption",
              ],
            },
          },
        ],
        "max_score": 7.637228,
        "total": {
          "relation": "eq",
          "value": 1,
        },
      },
      "status": 200,
      "timed_out": false,
      "took": 2,
    },
  ],
  "took": 2,
}
`;

exports[`API connector works as expected: Algolia request 1`] = `
[
  {
    "indexName": "imdb_movies",
    "params": {
      "facetFilters": [
        [
          "type:movie",
        ],
      ],
      "facets": [
        "*",
      ],
      "highlightPostTag": "</em>",
      "highlightPreTag": "<em>",
      "maxValuesPerFacet": 10,
      "page": 0,
      "query": "shawshank",
      "tagFilters": "",
    },
  },
  {
    "indexName": "imdb_movies",
    "params": {
      "analytics": false,
      "attributesToHighlight": [],
      "attributesToRetrieve": [],
      "attributesToSnippet": [],
      "clickAnalytics": false,
      "facets": "type",
      "highlightPostTag": "</em>",
      "highlightPreTag": "<em>",
      "hitsPerPage": 1,
      "maxValuesPerFacet": 10,
      "page": 0,
      "query": "shawshank",
      "tagFilters": "",
    },
  },
]
`;
